source/slang/slang-ir-specialize.cpp

```diff
@@ -1010,6 +1017,9 @@ struct SpecializationContext
                     workList.removeLast();
                     workListSet.remove(inst);
 
+                    if (!inst->getParent() && inst->getOp() != kIROp_Module)
```

Comment:
Yong's note: we can run into instructions that we previously add to work list but now have been removed. If the inst has been removed (we check this by seeing if the inst still has a parent), there is no need to process it anymore.